File: leetcode/2022/weekly_274/as_before_bs.cpp
#include <bits/stdc++.h>
#include <string.h>

using namespace std;

class Solution {
public:
  vector<size_t> allOccurrances(string needle, string haystack) {
    size_t pos = haystack.find(needle, 0);
    vector<size_t> positions;

    while (pos != string::npos) {
      positions.push_back(pos);
      pos = haystack.find(needle, pos + 1);
    }

    return positions;
  }

  bool checkString(string s) {
    vector<size_t> as = allOccurrances("a", s);
    vector<size_t> bs = allOccurrances("b", s);
  }
}

File: leetcode/2022/weekly_274/as_before_bs.py
class Solution:
    def checkString(self, s):
        az = [i for i in range(len(s)) if s.startswith("a", i)]
        bz = [i for i in range(len(s)) if s.startswith("b", i)]

        if len(az) == 0:
            return True

        last_a = az[-1]
        for b in bz:
            if last_a > b:
                return False
        return True
